src/main/java/com/jmeter/websocket/plugin/endpoint/WebsocketClient.java

```diff
@@ -0,0 +1,25 @@
+package com.jmeter.websocket.plugin.endpoint;
+
+import com.jmeter.websocket.plugin.endpoint.comsumers.WebsocketMessageProcessor;
+import org.apache.jmeter.protocol.http.control.CookieManager;
+import org.apache.jmeter.samplers.SampleResult;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+public interface WebsocketClient {
+
+    void connect(URI uri, CookieManager cookieManager, Map<String, List<String>> headers, SampleResult result, long timeOut) throws Exception;
```

Comment:
I prefer simpler solution for now as it is done in jmeter http request sampler: 
![image](https://cloud.githubusercontent.com/assets/17699437/22366110/32bbb47e-e47b-11e6-93e4-49dae4c059cb.png)